src/server/logger.js

```diff
@@ -1,27 +1,97 @@
 'use strict';
 
-var levels = ['trace', 'info', 'log', 'debug', 'warn', 'error'],
-    debug = require('debug');
+const chalk = require('chalk');
+
+const LEVELS = ['trace', 'info', 'log', 'debug', 'warn', 'error'],
+    LEVELCOLORS = {
+        'trace': {
+            bg: 'bgBlack',
+            fg: 'white'
+        }, 
+        'info': {
+            bg: 'bgBlack',
+            fg: 'white'
+        },
+        'log': {
+            bg: 'bgBlack',
+            fg: 'white'
+        },
+        'debug': {
+            bg: 'bgBlack',
+            fg: 'white'
+        },
+        'warn':  {
+            bg: 'bgYellow',
+            fg: 'black'
+        },
+        'error': {
+            bg: 'bgRed',
+            fg: 'white'
+        }
+    };
 
 // set which debug levels to send to stderr, the rest will go to stdout
 const STDERR = ['warn', 'error'];
 
 var Logger = function(name) {
     this._name = name;
-    /* eslint-disable no-console*/
-    levels.forEach(lvl => {
-        this[lvl] = debug(`${name}:${lvl}`);
-        if (STDERR.find(item => item === lvl)) {
-            this[lvl].log = console.error.bind(console);
-        } else { // send to stdout
-            this[lvl].log = console.log.bind(console);
-        }
+    LEVELS.forEach(lvl => {
+        this[lvl] = this._log.bind(this, lvl);
     });
+};
+
+Logger.prototype._log = function(level, content) {
+
+    // Allow skips
+    if(Logger.prototype.skips.some((skip) => skip.test(this._name)))
+    {
+        return;
+    }
+
+    // Allow whitelist
+    if(Logger.prototype.names.length > 0){
+        
+        if(!Logger.prototype.names.some((name) => name.test(this._name)))
+        {
+            return;
+        }
+    }
+
+    /* eslint-disable no-console*/
+    // Determine which output to use
+    let logFunc = STDERR.find(lvl => level === lvl)? console.error : console.log;
+    logFunc = logFunc.bind(console);
     /* eslint-enable no-console*/
+
+    // Prevent issues if no color available
+    logFunc(chalk[LEVELCOLORS[level].bg](chalk[LEVELCOLORS[level].fg](`${Date.now()} ${this._name}:${level}`)) + ' ' + content);
 };
 
 Logger.prototype.fork = function(name) {
     return new Logger([this._name, name].join(':'));
 };
 
+// Inspired by debug package https://github.com/visionmedia/debug/
+Logger.prototype.names = [];
+Logger.prototype.skips = [];
+
+// Get input from env variable
+let namespaces = process.env.DEBUG || '';
+
+let split = namespaces.split(/[\s,]+/);
```

Comment:
what is a namespace? in "netsblox:storage" is storage a namespace? if yes shouldn't those be separated by ':' instead of space? including ":" in an env variable is more straightforward that space (no need for escaping)